### PR TITLE
Change Object mode for package testing & remove older frameworks & rids

### PR DESCRIPTION
**Pull request by Anipik**
- use nuget content model for package object similar to package validation.
-  remove out of support frameworks.
- remove rids metadata as we no longer need.
-  modify tests

### CODE REVIEW

File: src/Microsoft.DotNet.PackageTesting/GetCompatiblePackageTargetFrameworks.cs [hunk 1]
```diff
@@ -32,69 +34,51 @@ public override bool Execute()
...
                 if (packageTfmMapping.ContainsKey(item))
-                    frameworksToTest.AddRange(packageTfmMapping[item].ToList());
+                    frameworksToTest.AddRange(packageTfmMapping[item]);
+                // Adding the frameworks in the packages to the test matrix.
+                frameworksToTest.Add(item);
             }
 
-            // Pruning the test matrix by removing the frameworks we dont want to test.
-            frameworksToTest = frameworksToTest.Where(tfm => allTargetFrameworks.Contains(tfm)).ToList();
+            if (!string.IsNullOrEmpty(minDotnetTargetFramework) && frameworksToTest.Contains(FrameworkConstants.CommonFrameworks.NetStandard20))
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.PackageTesting/GetCompatiblePackageTargetFrameworks.cs:67]: What about netstandard2.1?

File: src/Microsoft.DotNet.PackageTesting/GetCompatiblePackageTargetFrameworks.cs [hunk 2]
```diff
@@ -19,6 +19,8 @@ public class GetCompatiblePackageTargetFrameworks : BuildTask
         [Required]
         public string[] PackagePaths { get; set; }
 
+        public string MinDotnetTargetFramework { get; set; }
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.PackageTesting/GetCompatiblePackageTargetFrameworks.cs:22]: Is this property necessary? Can't the logic automatically find the lower .NETCoreApp version from the `allTargetFrameworks` list?

File: src/Microsoft.DotNet.PackageTesting/GetCompatiblePackageTargetFrameworks.cs [hunk 3]
```diff
@@ -32,69 +34,51 @@ public override bool Execute()
...
-            allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.NetCoreApp21);
-            allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.NetCoreApp30);
             allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.NetCoreApp31);
             allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.Net50);
-            allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.Net45);
-            allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.Net451);
-            allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.Net452);
-            allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.Net46);
             allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.Net461);
             allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.Net462);
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.PackageTesting/GetCompatiblePackageTargetFrameworks.cs:79]: Why are we missing net47, net471, net472, net48 and net6.0?

File: src/Microsoft.DotNet.PackageTesting/GetCompatiblePackageTargetFrameworks.cs [hunk 4]
```diff
@@ -32,69 +34,51 @@ public override bool Execute()
...
-            allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.NetStandard10);
-            allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.NetStandard11);
-            allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.NetStandard12);
-            allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.NetStandard13);
-            allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.NetStandard14);
-            allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.NetStandard15);
-            allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.NetStandard16);
-            allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.NetStandard17);
             allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.NetStandard20);
             allTargetFrameworks.Add(FrameworkConstants.CommonFrameworks.NetStandard21);
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.PackageTesting/GetCompatiblePackageTargetFrameworks.cs:81]: As already provided as feedback in the previous PR, can we avoid the static list and instead pass in an item that is defined in the package's props file? Imagine we want to drop support for a .NETFramework or .NETCoreApp tfm or add support for a new TFM in dotnet/runtime. That would require a package update. Making it possible to mutate the data from outside would make this easier.

**Comment by ViktorHofer** [comment]
Cool. You are missing the props file which defines the current set of tfms:
- netstandard2.0
- netstandard2.1
- net461-net48
- netcoreapp3.1
- net5.0
- net6.0

**Comment by Anipik** [comment]
> Cool. You are missing the props file which defines the current set of tfms:

shouldnt this be set in the runtime repo ? so that when the tfms  go out of support we can just change the property in the runtime repo.

**Comment by ViktorHofer** [comment]
> shouldnt this be set in the runtime repo ? so that when the tfms go out of support we can just change the property in the runtime repo.

Projects that live in arcade are meant to be consumed by multiple repositories. Moving the current set of supported TFMs into a props file a) makes it easier for multiple repositories to use the package and b) allows consumers to mutate that list.

Imagine a repository like dotnet/runtime that versions to the next version of .NETCoreApp which wants to do that independently of the current set of exposed supported TFMs in this package.

In example dotnet/runtime versions from .NET 6 to .NET 7 and can add this code block temporarily to avoid a change in the package itself:

```
<ItemGroup>
   <SupportedTargetFramework Remove="netcoreapp3.1" />
   <SupportedTargetFramework Remove="net5.0" />
   <SupportedTargetFramework Include="net7.0" />
</ItemGroup>
```

At a later point these changes will then flow back into this package to keep the defined TFMs consistent with the .NET support policy.

File: src/Microsoft.DotNet.PackageTesting/build/Microsoft.DotNet.PackageTesting.props [hunk 5]
```diff
@@ -3,4 +3,8 @@
     <DotNetPackageTestingAssembly Condition="'$(MSBuildRuntimeType)' != 'Core'">$(MSBuildThisFileDirectory)..\tools\net472\Microsoft.DotNet.PackageTesting.dll</DotNetPackageTestingAssembly>
     <DotNetPackageTestingAssembly Condition="'$(MSBuildRuntimeType)' == 'Core'">$(MSBuildThisFileDirectory)..\tools\netcoreapp3.1\Microsoft.DotNet.PackageTesting.dll</DotNetPackageTestingAssembly>
   </PropertyGroup>
+
+  <ItemGroup>
+    <SupportedTestFramework Include="netcoreapp3.1;net5.0;net6.0;net461;net462;net471;net472;net48;netstandard2.0;netstandard2.1" />
```
Comment (Anipik) [inline src/Microsoft.DotNet.PackageTesting/build/Microsoft.DotNet.PackageTesting.props:8]: SupportedTargetFramework is currently set by the sdk and it contains all the frameworks so i defined it as ```SupportedTestFramework```

Comment (ViktorHofer) [inline src/Microsoft.DotNet.PackageTesting/build/Microsoft.DotNet.PackageTesting.props:8]: Just for better visibility, can you please break them apart and Include them one by one and group them by TFI?

Comment (ViktorHofer) [inline src/Microsoft.DotNet.PackageTesting/build/Microsoft.DotNet.PackageTesting.props:8]: I believe you are missing test coverage in the test assembly as when I tried to add net6.0 last time, the NuGet Parse mechanism blew up. Can you please add a test that passes these items in?

Comment (Anipik) [reply src/Microsoft.DotNet.PackageTesting/build/Microsoft.DotNet.PackageTesting.props:8]: the tests contains 

https://github.com/dotnet/arcade/pull/7510/files#diff-8146571ca49951a4551bb5598cead7b8edde7163a3282710165078471ccbd754R14

and tested with runtime repo as well.

**Review by ViktorHofer — APPROVED** [review]
Thanks a lot 👍

**Pull request merged**